merge_sort/merge_sort.c

```diff
@@ -0,0 +1,57 @@
+#include <stdio.h>
+
+void merge(int *arr, int left, int mid, int right) {
+	int i, j, k;
+	int n_left = mid - left + 1;
+	int n_right = right - mid;
+	int left_arr[n_left];
+	int right_arr[n_right];
+	for (i = 0; i < n_left; i++)
+		left_arr[i] = arr[left + i];
+	for (j = 0; j < n_right; j++)
+		right_arr[j] = arr[mid + 1 + j];
+	i = 0;
+	j = 0;
+	k = left;
+	while (i < n_left && j < n_right) {
+		if (left_arr[i] <= right_arr[j]) {
+			arr[k] = left_arr[i];
+			i++;
+		}
+		else {
```

Comment:
@singhpratyush It wasn't pointed out in previous algorithms so I thought we aren't following that rule. I'll make the necessary changes.